docs/source/deploy_groups.rst

```diff
@@ -0,0 +1,82 @@
+
+.. sourcecode:: yaml
+
+   ---
+   pipeline:
+   - instancename: itest
+   - instancename: security-check
+   - instancename: push-to-registry
+   - instancename: performance-check
+   - instancename: dev-stage.everything
+     trigger_next_step_manually: true
+   - instancename: prod.canary
+     trigger_next_step_manually: true
+   - instancename: prod.non_canary
+
+You can see we have three deploy groups here: ``dev-stage.everything``, ``prod.canary`` and ``prod.non_canary``. There each refer to a number of instances that all share the same docker container, and will all be deployed at the same time.
+
+Now let’s take a look at how instances are linked to a deploy group by taking a look in a sample config, ``marathon-norcal-devc.yaml``:
+
+.. sourcecode:: yaml
+
+   ---
+   canary:
+     cpus: .1
+     mem: 301
+     nerve_ns: main
+     instances: 1
+     deploy_group: dev-stage.everything
+   main:
+     cpus: 1
+     mem: 300
+     instances: 3
+     deploy_groups: dev-stage.everything
+
+This is repeated for each ``marathon-*.yaml`` and ``chronos-*.yaml`` file. In example_service, we create three different files - ``marathon-PROD.yaml``, ``marathon-STAGE.yaml`` and ``marathon-DEV.yaml`` and use symlinks to create the files we need. This is similar to the current ``marathon-SHARED.yaml`` file. Feel free to look at example_service in ``yelpsoa_configs`` for an example on how to use deploy groups.
+
+If no value for ``deploy_group`` is given, it defaults to ``CLUSTER_NAME.INSTANCE_NAME``. This is to support legacy services that don't use deploy groups.
+
+What deploy groups aren't
+-------------------------
+
+Deploy groups are not a way to control your instances -- tools such as ``paasta start/stop/restart`` do not support deploy groups: you can’t stop all an entire deploy group by running ``paasta stop DEPLOY_GROUP_NAME``. Deploy groups also do not share other configuration elements such as memory and cpu allocation.
+
+How to use deploy groups
+========================
+
+New service
+-----------
+
+When you run ``paasta fsm`` to generate the boilerplate configs for your paasta service, the default configs will have four deploy groups: ``dev.everything``, ``stage.everything``, ``prod.canary`` and ``prod.non_canary``. You can change these deploy groups around to suit your needs, and also change their deploy order by editing ``deploy.yaml``. Then, edit your soa configs and generate your pipline like normal.
+
+Existing service
+----------------
+
+Edit your ``deploy.yaml`` and your ``marathon-*.yaml`` and ``chronos-*.yaml`` files to use deploy groups. For each instance in a deploy group, specify a ``deploy_group`` parameter in their config file. Then, remove their deployment steps from ``deploy.yaml`` and replace it with a single deploy step for the deploy group. Finally, update your jenkins workflow by deleting and re-creating it like normal.
```

Comment:
Are you sure this order is correct? Seems like the proper order is:
* Add deploy steps in deploy.yaml
* Run the pipeline to mark them
* Edit marathon/chronos yamls and remove old steps from deploy.yaml